src/components/forms/NetworkDevicesForm/NetworkDeviceIPAddress.tsx

```diff
@@ -0,0 +1,76 @@
+import { type FC } from "react";
+import type { InstanceAndProfileFormikProps } from "../instanceAndProfileFormValues";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { Input } from "@canonical/react-components";
+import { type LxdNetwork } from "types/network";
+import { type LxdNicDevice } from "types/device";
+import { typesWithNicStaticIPSupport } from "util/networks";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  index: number;
+  network?: LxdNetwork;
+  device: LxdNicDevice;
+  family: "IPv4" | "IPv6";
+}
+
+export const NetworkDeviceIPAddress: FC<Props> = ({
+  formik,
+  index,
+  network,
+  device,
+  family,
+}) => {
+  if (!network || !network.config) {
+    return null;
+  }
+
+  const readOnly = (formik.values as EditInstanceFormValues).readOnly;
+  const addressString = family === "IPv4" ? "ipv4.address" : "ipv6.address";
+  const networkIP = network.config[addressString];
+  const deviceIP = device[addressString];
+  const isStaticIPAllowed =
+    network.managed && typesWithNicStaticIPSupport.includes(network.type);
+  const canSelectIP = isStaticIPAllowed && networkIP && networkIP !== "none";
+
+  if (readOnly) {
+    return (
+      <div className="ip-content">
+        <>
+          <div>{family}</div>
+          <div className="mono-font">
+            <b>{deviceIP || "Dynamic"}</b>
+          </div>
+        </>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <Input
+        id={`${addressString}-${index}`}
+        name={`${addressString}-${index}`}
+        label={`${family} address reservation`}
+        type="text"
+        className="ip-content-edit"
```

Comment:
This is applying the class to the input, we'd need to have it on a wrapper around the label and help text.

Current:

<img width="379" height="594" alt="image" src="https://github.com/user-attachments/assets/ce72e501-ed92-4c8c-8a24-050b7f302c5b" />